src/components/subscription-plans/blocks/payments-table.tsx

```diff
@@ -0,0 +1,200 @@
+"use client";
+
+import { ShortAddress } from "@/components/short-address";
+import { Card, CardContent } from "@/components/ui/card";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { formatCurrencyLabel } from "@/lib/constants/currencies";
+import type { SubscriptionPayment } from "@/lib/types";
+import type { SubscriptionPlan } from "@/server/db/schema";
+import { api } from "@/trpc/react";
+import { format } from "date-fns";
+import {
+  CreditCard,
+  DollarSign,
+  ExternalLink,
+  Filter,
+  Receipt,
+} from "lucide-react";
+import { useState } from "react";
+import { EmptyState } from "../../dashboard/blocks/empty-state";
+import { Pagination } from "../../dashboard/blocks/pagination";
+import { StatCard } from "../../dashboard/blocks/stat-card";
+import { TableHeadCell } from "../../dashboard/blocks/table-head-cell";
+
+const ITEMS_PER_PAGE = 10;
+
+interface PaymentsTableProps {
+  initialPayments: SubscriptionPayment[];
+  subscriptionPlans: SubscriptionPlan[];
+}
+
+const PaymentTableColumns = () => (
+  <TableRow className="hover:bg-transparent border-none">
+    <TableHeadCell>Payment Date</TableHeadCell>
+    <TableHeadCell>Plan Name</TableHeadCell>
+    <TableHeadCell>Currency</TableHeadCell>
+    <TableHeadCell>Amount</TableHeadCell>
+    <TableHeadCell>Subscriber</TableHeadCell>
+    <TableHeadCell>Request Scan URL</TableHeadCell>
+  </TableRow>
+);
+
+const PaymentRow = ({ payment }: { payment: SubscriptionPayment }) => {
+  return (
+    <TableRow className="hover:bg-zinc-50/50">
+      <TableCell>
+        {format(new Date(payment.createdAt), "do MMM yyyy")}
+      </TableCell>
+      <TableCell className="font-medium">{payment.planName}</TableCell>
+      <TableCell>{formatCurrencyLabel(payment.currency)}</TableCell>
+      <TableCell>
+        <span className="font-semibold">
+          {Number(payment.amount).toLocaleString()} ${payment.currency}
+        </span>
```

Comment:
Incorrect Currency Display Format: You're showing "$100 $USD" instead of the proper "$100 USD" or "100 USD" format. The dollar sign conflicts with non-USD currencies.
```suggestion
        <span className="font-semibold">
          {Number(payment.amount).toLocaleString()} {payment.currency}
        </span>
```